observer/monitor.go

```diff
@@ -0,0 +1,59 @@
+package observer
+
+import (
+	"time"
+
+	blog "github.com/letsencrypt/boulder/log"
+	p "github.com/letsencrypt/boulder/observer/probes"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// monitor contains the parsed, normalized, and validated configuration
+// describing a given oberver monitor
+type monitor struct {
+	name   string
+	period time.Duration
+	probe  string
+	prober p.Prober
+	logger blog.Logger
+	metric prometheus.Registerer
+}
+
+// start creates a ticker channel then spins off a prober goroutine for
+// each period specified in the monitor config and a timeout inferred
+// from that period. This is not perfect, it means that the effective
+// deadline for a prober goroutine will be TTL + time-to-schedule, but
+// it's close enough for our purposes
+func (m monitor) start() *time.Ticker {
+	ticker := time.NewTicker(m.period)
+	logInfo := func(status string, took string) {
```

Comment:
This function appears to only be a closure over `m`, the receiver, so it could just be another method with the same receiver instead.

Reply:
I pass the status and the receiver as params. Maybe I am missing something?

Reply:
Ah, nevermind, I think I understand you now.